wooting-macro-backend/src/plugin/system_event.rs

```diff
@@ -35,263 +26,102 @@ pub enum DirectoryAction {
                 VolumeAction::LowerVolume => {
-                    util::send_key(&send_channel, vec![*SCANCODE_TO_RDEV.get(&0x81).unwrap()])
-                        .await;
+                    util::direct_send_key(&send_channel, vec![rdev::Key::VolumeDown]).await?;
                 }
                 VolumeAction::IncreaseVolume => {
-                    util::send_key(&send_channel, vec![*SCANCODE_TO_RDEV.get(&0x80).unwrap()])
-                        .await;
+                    util::direct_send_key(&send_channel, vec![rdev::Key::VolumeUp]).await?;
                 }
             },
-
-            SystemAction::Brightness { action } => {
-                // This is split for windows/linux and macOS.
-                // macOS cannot implement this feature, so due to language limitations we have to make two separate matches.
-                #[cfg(any(target_os = "windows", target_os = "linux"))]
-                match action {
-                    MonitorBrightnessAction::SetAll { level } => {
-                        #[cfg(any(target_os = "windows", target_os = "linux"))]
-                        brightness_set_all_device(*level).await;
-                        #[cfg(target_os = "macos")]
-                        error!("Not supported on macOS");
-                    }
-                    MonitorBrightnessAction::SetSpecific { level, name } => {
-                        #[cfg(any(target_os = "windows", target_os = "linux"))]
-                        brightness_set_specific_device(*level, name).await;
-                        #[cfg(target_os = "macos")]
-                        error!("Not supported on macOS");
-                    }
-                    MonitorBrightnessAction::ChangeSpecific { by_how_much, name } => {
-                        #[cfg(any(target_os = "windows", target_os = "linux"))]
-                        brightness_change_specific(*by_how_much, name).await;
-                        #[cfg(target_os = "macos")]
-                        error!("Not supported on macOS");
-                    }
-                    MonitorBrightnessAction::ChangeAll { by_how_much } => {
-                        #[cfg(any(target_os = "windows", target_os = "linux"))]
-                        brightness_change_all(*by_how_much).await;
-                        #[cfg(target_os = "macos")]
-                        error!("Not supported on macOS");
-                    }
-                }
-                // Second match for macOS
-                #[cfg(target_os = "macos")]
-                match action {
-                    _ => {
-                        error!("Not supported on macOS");
-                    }
-                }
-            }
-
             SystemAction::Clipboard { action } => match action {
                 ClipboardAction::SetClipboard { data } => {
                     ClipboardContext::new()
-                        .unwrap()
+                        .map_err(|err| anyhow::Error::msg(err.to_string()))?
```

Comment:
need to map errors to string to pass them properly. ClipboardContext crate doesn't have the proper error conversions needed for anyhow

Reply:
There's defo no cleaner way? It's pretty nasty having all these conversion everywhere

Reply:
Since the error type doesn't satisfy the constraints, I have to create a new one with the message and convert. Maybe there is a cleaner way, but nothing I tried at least worked properly